### PR TITLE
feat(core): support block or element reference links

### PR DESCRIPTION
**Pull request by fundon**
Upstreams: https://github.com/toeverything/blocksuite/pull/8021

* open doc with mode
* monitor mode changes in query string
* scroll anchoring

https://github.com/user-attachments/assets/681abff8-e51b-47ea-bb71-447e8b312142

https://github.com/user-attachments/assets/e73ed4c0-4e33-45f8-9db4-d8eed3525d05

### CODE REVIEW

**Comment by fundon** [comment]
* **#7946** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7946?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 3  dependent PRs ([#8001](https://github.com/toeverything/AFFiNE/pull/8001) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8001?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8002](https://github.com/toeverything/AFFiNE/pull/8002) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8002?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#8056](https://github.com/toeverything/AFFiNE/pull/8056) <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8056?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈
* **#7933** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7933?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7886** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7886?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts [hunk 1]
```diff
@@ -46,21 +53,23 @@ function createCopyLinkToBlockMenuItem(
     name: 'Copy link to block',
     icon: LinkIcon({ width: '20', height: '20' }),
     hide: () => {
-      const { editor } = framework.get(EditorService);
-      const mode = editor.mode$.value;
+      const pageId = model.doc.id;
+      const docsService = framework.get(DocsService);
+      const mode = docsService.list.getPrimaryMode(pageId) || DocMode.Page;
```
Comment (EYHN) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts]: PrimaryMode !== editor mode

Comment (EYHN) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts]: should use editor mode here?

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/widgets/toolbar.ts [hunk 2]
```diff
@@ -76,10 +79,11 @@ function createCopyLinkToBlockMenuItem(
...
       if (!baseUrl) return;
 
-      const { editor } = framework.get(EditorService);
-      const mode = editor.mode$.value;
-      const pageId = editor.doc.id;
-      const workspaceId = editor.doc.workspace.id;
+      const pageId = ctx.doc.id;
+      const workspace = framework.get(WorkspaceService).workspace;
+      const docsService = framework.get(DocsService);
+      const mode = docsService.list.getPrimaryMode(pageId) || DocMode.Page;
```
Comment (EYHN) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/widgets/toolbar.ts]: should use editor mode here?

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts (hunk 1 above)
Comment (fundon) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts]: Try it before, `EditorService` will be missing in center peek

Comment (fundon) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/specs/custom/database-block.ts]: Let me try it again.

**Comment by fundon** [comment]
@pengx17  Missing `EditorService` in center peek


https://github.com/user-attachments/assets/47f004e7-447b-49df-9849-c3305add2f53

**Review by forehalo — APPROVED** [review]

**Review by EYHN — APPROVED** [review]

**Pull request merged**